app/policies/repository_policy.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+# Policies for RepositoriesController
+class RepositoryPolicy < ApplicationPolicy
+  attr_reader :current_user, :repository
+
+  def initialize(current_user, repository)
+    @current_user = current_user
+    @repository = repository
+    super
+  end
+
+  def show?
+    repository.public_access == true ||
+      !!current_user&.accessible_repositories&.map(&:to_param)&.include?(repository.to_param)
+  end
+
+  def create?(owner)
+    if owner.is_a?(User)
```

Comment:
Maybe do a `return false unless current_user` at the beginning of this method instead of inside the if, so that it's easier to see that you need to be logged in no matter what